src/js/stream_writable.js

```diff
@@ -37,7 +37,9 @@ function WritableState(options) {
 
   // high water mark.
   // The point where write() starts retuning false.
-  var hwm = options.highWaterMark;
+  var hwm = defaultHighWaterMark;
+  if(options)
+    hwm = options.highWaterMark;
   this.highWaterMark = (hwm || hwm === 0) ? hwm : defaultHighWaterMark;
```

Comment:
I suggest implementing the above 4 lines as below.
``` js
this.highWaterMark = (options && util.isNumber(options.highWaterMask)) ? 
  options.highWaterMask : defaultHighWaterMask;
```
And space is required between `if` and `(`.